headers/validation.h

```diff
@@ -222,14 +283,25 @@ struct NormalDistributionGenerator {
 
 	const T mean_, stddev_;
 
-	explicit NormalDistributionGenerator(T mean, T stddev) : mean_(mean), stddev_(stddev) {}
+	explicit NormalDistributionGenerator(T mean, T stddev) : mean_(mean), stddev_(stddev) {
+		assert(stddev_ >= 0);
+	}
 
 	// NOTE: Currently this retries instead of clamping to the interval.
 	T operator()(T low, T high, std::mt19937_64& rng) const {
+		assert(low < high);
 		T v;
-		do {
-			v = std::normal_distribution<T>(mean_, stddev_)(rng);
-		} while(v < low or v > high);
+		while (true) {
+			T u1 = Random::real64(rng);
+			T u2 = Random::real64(rng);
+			// Box-Muller-Methode
```

Comment:
Can you add a link/reference?

Reply:
https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform